core/src/main/java/org/eclipse/hono/util/TenantObject.java

```diff
@@ -641,4 +642,168 @@ public OptionalInt getTraceSamplingPriority(final String authId) {
+     */
+    @JsonIgnore
+    public Set<X500Principal> getTrustedCaSubjectDns() {
+
+        if (trustConfigurations == null) {
+            return null;
+        } else {
+            final Set<X500Principal> subjectNames = new HashSet<>();
+            trustConfigurations.keySet().forEach(subjectDn -> {
+                subjectNames.add(new X500Principal(subjectDn));
+            });
+            return subjectNames;
+        }
+    }
+
+    /**
+     * Get the list of configured trusted CAs for this tenant.
+     * 
+     * @return The list of configured trusted CAs or {@code null} if this tenant has no configured trusted CA.
+     *
+     */
+    @JsonIgnore
+    public List<JsonObject> getTrustedCAs() {
+
+        if (trustConfigurations == null) {
+            return null;
+        } else {
+            return trustConfigurations.values().stream().flatMap(Collection::stream).collect(Collectors.toList());
+        }
+
+    }
+
+    @JsonIgnore
+    private TrustAnchor getTrustAnchor(final JsonObject trustedCa) throws GeneralSecurityException {
+        final X509Certificate cert = getTrustedCertificateAuthority(trustedCa);
+        if (cert != null) {
+            return new TrustAnchor(cert, null);
+        } else {
+            return getTrustAnchorForPublicKey(trustedCa);
+        }
+    }
+
+    /**
+     * Get the X.509 certificate from the specified trusted CA configuration.
+     * 
+     * @param trustedCa The trusted CA configuration to construct an X.509 certificate from.
+     * 
+     * @return The certificate or {@code null} if no certificate authority
+     *          has been set or the validity period of the certificate is missing.
+     * @throws CertificateException if the value of <em>cert</em> property cannot be parsed into
+     *              an X.509 certificate.
+     */
+    @JsonIgnore
+    private X509Certificate getTrustedCertificateAuthority(final JsonObject trustedCa) throws CertificateException {
+
+        final String obj = (String) trustedCa.getValue(TenantConstants.FIELD_PAYLOAD_CERT);
+        if (obj == null) {
+            return null;
+        } else {
+            try {
+                final byte[] derEncodedCert = Base64.getDecoder().decode(((String) obj));
+                final CertificateFactory factory = CertificateFactory.getInstance("X.509");
+                return (X509Certificate) factory.generateCertificate(new ByteArrayInputStream(derEncodedCert));
+            } catch (final IllegalArgumentException e) {
+                // ignore -> should never happen
+                throw new CertificateParsingException("cert property does not contain valid Base64 encoding scheme", e);
+            }
+        }
+    }
+
+    /**
+     * Add a trusted CA configuration for this tenant.
+     * 
+     * @param trustedCa The trust configuration to add.
+     * 
+     * @return This tenant so the API can be used fluently.
+     */
+    public TenantObject addTrustedCA(final JsonObject trustedCa) {
```

Comment:
is this really supposed to be public? Who else needs this method?

Reply:
good catch